src/prng.c

```diff
@@ -17,38 +17,42 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <assert.h>
+
 #include "nwipe.h"
 #include "prng.h"
 #include "context.h"
 #include "logging.h"
 
 #include "mt19937ar-cok/mt19937ar-cok.h"
 #include "isaac_rand/isaac_rand.h"
+#include "isaac_rand/isaac64.h"
 
 nwipe_prng_t nwipe_twister = { "Mersenne Twister (mt19937ar-cok)", nwipe_twister_init, nwipe_twister_read };
 
 nwipe_prng_t nwipe_isaac = { "ISAAC (rand.c 20010626)", nwipe_isaac_init, nwipe_isaac_read };
+nwipe_prng_t nwipe_isaac64 = { "ISAAC-64 (isaac64.c)", nwipe_isaac64_init, nwipe_isaac64_read };
 
-int nwipe_u32tobuffer( u8* buffer, u32 rand, int len )
+/* Print given number of bytes from unsigned integer number to a byte stream buffer starting with low-endian. */
+static void nwipe_u32tobuffer( u8* buffer, u32 val, int len )
 {
-    /*
-     * Print given number of bytes from unsigned integer number to a byte stream buffer starting with low-endian.
-     */
+    assert( len <= sizeof(u32) );
```

Comment:
Although assert is more succinct, I'd prefer all errors/warning/info to be send to nwipe_log as it makes it easier for users to just send the log file when reporting a bug or issue and when then know that all errors/warning will be in the log with timestamps.

Reply:
Shouldn't be a problem, though I figured assert() used this way is a neat way to document that the condition is guaranteed never to occur (in this case, because of the modulo before). I think this is an older commit though, the later one moved the asserts down.

Reply:
Other than the assert, looks good. I've run twelve tests on a 1GB loop drive to give me some idea of the relative speed improvement and it's definitely an improvement and justifies having Isaac64 as the default PRNG.
| Zero | Isaac64 | Isaac(32) | Mersenne |
| ------ | ----------- | ------------- | ------------- |
| 11 seconds | 14 seconds | 15 seconds | 16s |
| 11 seconds | 14 seconds | 15 seconds | 16s |
| 11 seconds | 14 seconds | 16 seconds | 16s |

Reply:
I suspect Isaac64 won't perform well on a 32-bit system, so on those it should default to 32-bit ISAAC.